blobfuse/include/FileLockMap.h

```diff
@@ -20,7 +20,8 @@ class file_lock_map
 public:
     static file_lock_map* get_instance();
     std::shared_ptr<std::mutex> get_mutex(const std::string& path);
-
```

Comment:
It is risky to obtain multiple mutexs at the same time. Specify a lock order for the mutexs

Reply:
There are two mutexes. One is applied only to block download and deletion of file. As per existing code when file is under download we are not allowed to delete it. In case of delayed download the file locks will be releases as soon as open call returns. However we still need to block read/write calls as the file download may be going in background. Hence we have added another level of locks just to mark that file download is still going on in background and do not allow read/write.